GitHubMilestones.ps1

```diff
@@ -0,0 +1,506 @@
+        [Parameter(Mandatory, ParameterSetName='RepositoryElements')]
+        [string] $OwnerName,
+
+        [Parameter(Mandatory, ParameterSetName='MilestoneElements')]
+        [Parameter(Mandatory, ParameterSetName='RepositoryElements')]
+        [string] $RepositoryName,
+
+        [Parameter(Mandatory, ParameterSetName='MilestoneUri')]
+        [Parameter(Mandatory, ParameterSetName='RepositoryUri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName='MilestoneUri')]
+        [Parameter(Mandatory, ParameterSetName='MilestoneElements')]
+        [int] $Milestone,
+
+        [Parameter(ParameterSetName='RepositoryUri')]
+        [Parameter(ParameterSetName='RepositoryElements')]
+        [ValidateSet('Open', 'Closed', 'All')]
+        [string] $State,
+
+        [Parameter(ParameterSetName='RepositoryUri')]
+        [Parameter(ParameterSetName='RepositoryElements')]
+        [ValidateSet('DueOn', 'Completeness')]
+        [string] $Sort,
+
+        [Parameter(ParameterSetName='RepositoryUri')]
+        [Parameter(ParameterSetName='RepositoryElements')]
+        [ValidateSet('Ascending', 'Descending')]
+        [string] $Direction,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+        'ProvidedMilestone' = $PSBoundParameters.ContainsKey('Milestone')
+    }
+
+    if ($PSBoundParameters.ContainsKey('Milestone'))
+    {
+        $uriFragment = "repos/$OwnerName/$RepositoryName/milestones/$Milestone"
+        $description = "Getting milestone $Milestone for $RepositoryName"
+    }
+    else
+    {
+        $getParams = @()
+
+        if ($PSBoundParameters.ContainsKey('Sort'))
+        {
+            if ($Sort -eq "Completeness")
+            {
+                $getParams += "sort=completeness"
+            }
+            elseif ($Sort -eq "DueOn")
+            {
+                $getParams += "sort=due_on"
+            }
+        }
+
+        if ($PSBoundParameters.ContainsKey('Direction'))
+        {
+            if ($Direction -eq "Ascending")
+            {
+                $getParams += "direction=asc"
+            }
+            elseif ($Direction -eq "Descending")
+            {
+                $getParams += "direction=desc"
+            }
+        }
+
+        if ($PSBoundParameters.ContainsKey('State'))
+        {
+            $State = $State.ToLower()
+            $getParams += "state=$State"
+        }
+
+        $uriFragment = "repos/$OwnerName/$RepositoryName/milestones`?" +  ($getParams -join '&')
+        $description = "Getting milestones for $RepositoryName"
+    }
+
+    $params = @{
+        'UriFragment' = $uriFragment
```

Comment:
It looks like you deleted this repo (probably by accident?) so I'm unable to clone this change and test it for myself.  I think I would have expected this to trigger a [PSScriptAnalyzer error](https://github.com/PowerShell/PowerShellForGitHub/blob/master/CONTRIBUTING.md#static-analysis) for referencing a variable that isn't _strictly_ defined in the same scope.  You'd have to first define/reference `$uriFragment` and `$description` in the same scope as this reference (as opposed to within the nested `if/then/else`.  While this does _work_ due to the way that PowerShell operates, it would be better to have the variable effectively declared in the same scope (so, outside the scope of the `if/then/else`).

Reply:
I've initialized the two variables to  [String]::Empty  above the if/else. Not sure what you mean by the first sentence of your comment though? Which repo? It works for me..

Reply:
ok I think i understand your comment. I created a branch instead of forking for this PR. Should I fork instead in the future?